packages/mgt-chat/src/utils/rewriteEmojiContent.ts

```diff
@@ -30,3 +30,68 @@ const processEmojiContent = (messageContent: string): string => {
  */
 export const rewriteEmojiContent = (content: string): string =>
   emojiMatch(content) ? processEmojiContent(content) : content;
+
+/**
+ * The emoji HTML string that an <emoji> tag is translated to.
+ *
+ * @param emoji used as the alt value.
+ * @param title used in titles.
+ * @param id used in URLs and id attributes,
+ * @param hasOtherContent determines which size of the emoji to fetch. default 50px.
+ * @returns HTML string.
+ */
+const emojiHtmlString = (emoji: string, title: string, id: string, hasOtherContent?: boolean): string => {
+  const size = hasOtherContent ? '20' : '50'; // 20px with content, 50px without.
+  return `<span contenteditable="false" title="${title}" type="(${id})" class="animated-emoticon-${size}-${id}"><img itemscope="" itemtype="http://schema.skype.com/Emoji" itemid="${id}" src="https://statics.teams.cdn.office.net/evergreen-assets/personal-expressions/v2/assets/emoticons/${id}/default/${size}_f.png" title="${title}" alt="${emoji}" style="width:${size}px;height:${size}px;"/></span>`;
+};
+
+/**
+ * Matches the emoji tag with id in group 2, alt in group 3 and title in group 4.
+ */
+const emojiRegexForHtml =
```

Comment:
The weakness of this regex is that it's dependent on the order of the attributes in the html string.

We should seek assurance from Teams that they won't be changing the attribute order on these, or alternately we build an emoji web component that does this binding logic for us.

@sebastienlevert thoughts here?

Reply:
From a perf standpoint, could we use the regex to identify regexes and then query them like other DOM elements? Would that be massively under-perf or would stay within the limits of the acceptable?

Reply:
Or... using the DOM elements approach that @sebastienlevert has suggested, we can do away with regex, parse the content from string and have something [like this](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAEzgCSgWwDYGUoBOAFAJSIDeAUIohAgM5SIAWW2iAvIgAYA8ADgD40AU2zY4iKMxEERiXiMxwAVjEQxkHAER042bYgCG2KDsC8G4Dg9w7CjYROgMJx92wbwD0S1TEGIAkojAImJSMHLIAHQK3moaWtoAjMAAnABMRgD6AF5GYACewEYQIoYmZtqAfBuAVLs2MHYO2gBaeflBxaXuXspqXULc1LQMTPxGBPSynIhgIgDuiAAiAPIAsgAKYxPEJIN0YIwocJhTo+OykacTAGIER-gEMGAA5kSsOAA0iADkUCIAHlAPG9sF8djQ9gdYjB6FNUJhIgBHECyfK4MQiaBwAgAQXERC+UNBlEGwCxRAhTChiDgwEQUPoJCoNHBw3iUyhkSeIig2KghBgACMQL8iNpNNoyAB+SXfL4AbkGLP2THK7J6ME53N5-KFIu05QliGlsoVzKGyrCDTVPk1PL5D11IlFtnshuNX3lxLNFMQ9FGSC4qAgIEwIjAUEiEDkRl+AFF7KHw-i-XlQabmSmwJEJnadcKnV89r9w7HkPUjAL7F9Pl8itgJkSzZns1r7YL8-iXSJq5b7GCM-6W7mHR2fvl+N3PtwiAASciaAC+JG4-ZozZz2pHIsL2CM9HoPa+eRgmBjImQAFpvLA9heAKwABgvehBO0V5oOJ6esLgwcTEajEQz3jJQwygUUvwld8vyHTd223eolHoOgJ0PRtmRgjc20dfFEMwKBx0nRBtFYKB+AALg8DxkJkU9swAa0IyMjg8WN1Sgs1MNbPMEN+TBNB7LtVw0TAnlg7DR3oAgIB7Ei+X4ehKOoqAYxgCB6EiX4jEwDSIGQLMaWANSREiGZAREAA3WQnjkMMLz3HN6A8CdxgQEwrz+fg5H3eB9g8Cy0g8BzuSc681IYDwXw8ZARCKEBTA8R9MmAC5ng4jDRPEniCy7QT6j7aDMqw7L8XKHtymErjh3ggtGHyKtPm0WZNGkcjH34P45RkGAnlYNqHw6uUoPfZsjH4CcwGQRxmBgbBkCIL9hI5OR+F3EoAHV6mYIhM37BddgYfQTIkF44UiAU4GQfJIkeGYCDQAAVFYABkdn2yhUAwHB7lIYkgA)


```typescript
function doHtmlStr() {
  const html = `<p>Hello there <emoji id="cool" alt="😎" title="Cool"></emoji> I feel tired. <emoji id="1f92a_zanyface" alt="🤪" title="Zany face"></emoji></p>`;
  const parser = new DOMParser();
  const dom = parser.parseFromString(html, 'text/html');
  const emojis = dom.querySelectorAll('emoji');

  for (const emoji of emojis) {
    const id = emoji.getAttribute('id') ?? '';
    const alt = emoji.getAttribute('alt') ?? '';
    const title = emoji.getAttribute('title') ?? '';

    const span = document.createElement('span');
    span.setAttribute('contentEditable', 'false');
    span.setAttribute('title', title);
    span.setAttribute('type', `(${id})`);
    span.setAttribute('class', 'animated-emoticon-50-cool');

    const img = document.createElement('img');
    img.setAttribute('itemscope', '');
    img.setAttribute('itemtype', 'http://schema.skype.com/Emoji');
    img.setAttribute('itemid', title);
    img.setAttribute(
      'src',
      'https://statics.teams.cdn.office.net/evergreen-assets/personal-expressions/v2/assets/emoticons/cool/default/50_f.png'
    );
    img.setAttribute('title', title);
    img.setAttribute('alt', alt);
    img.setAttribute('style', 'width:50px;height:50px;');

    span.appendChild(img);
    emoji.replaceWith(span);
  }
  console.log(dom.body.innerHTML);
}

doHtmlStr();

```